server/src/lib/stats.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright 2022 Oxide Computer Company
+use anyhow::anyhow;
+use dropshot::{ConfigDropshot, ConfigLogging, ConfigLoggingLevel};
+use omicron_common::api::internal::nexus::ProducerEndpoint;
+use oximeter::{
+    types::{Cumulative, Sample},
+    Metric, MetricsError, Producer, Target,
+};
+use oximeter_producer::{Config, Server};
+use slog::{error, info, Logger};
+
+use std::net::SocketAddr;
+use std::sync::{Arc, Mutex};
+use uuid::Uuid;
+
+// These structs are used to construct the desired stats for Oximeter.
+#[derive(Debug, Copy, Clone, Target)]
+struct InstanceUuid {
+    pub uuid: Uuid,
+}
+#[derive(Debug, Default, Copy, Clone, Metric)]
+pub struct Rebooted {
```

Comment:
Nit: this excludes guest OS-initiated reboots, so at some point we might want to clarify the name of this metric. Not worth blocking this PR for, but maybe worth filing a tracking issue for deciding exactly what server-wide metrics we want to export & making sure they're labeled/computed properly.

Reply:
Yeah, that's a good point about not seeing OS level reboots.  All this metric is really counting is how many times someone called the "/instance/state" endpoint with the "Reset" state. 

I was hoping with these changes to lay the groundwork that myself or others could come back and build on to create more meaningful stats inside Propolis.  These changes just open the door.  I tried to pick something that was both simple and easy to see change.  But, yeah, I agree the name here is not the best choice.

Reply:
https://github.com/oxidecomputer/propolis/issues/153

Reply:
I also updated the name in 395d4db.